pkg/sql/opt/norm/testdata/rules/limit

```diff
@@ -109,6 +109,61 @@ limit
  │    └── limit hint: 1.00
  └── -1
 
+# --------------------------------------------------
+# EliminateLimitInExists
+# --------------------------------------------------
+norm expect=EliminateLimitInExists
+SELECT * FROM a WHERE EXISTS (SELECT * FROM b where b.y = a.k LIMIT 1)
+----
+semi-join (hash)
+ ├── columns: k:1!null i:2 f:3 s:4 j:5
+ ├── key: (1)
+ ├── fd: (1)-->(2-5)
+ ├── scan a
+ │    ├── columns: k:1!null i:2 f:3 s:4 j:5
+ │    ├── key: (1)
+ │    └── fd: (1)-->(2-5)
+ ├── scan b
+ │    └── columns: y:7
+ └── filters
+      └── y:7 = k:1 [outer=(1,7), constraints=(/1: (/NULL - ]; /7: (/NULL - ]), fd=(1)==(7), (7)==(1)]
+
+norm expect=EliminateLimitInExists
+SELECT * FROM a WHERE NOT EXISTS (SELECT * FROM b where b.y = a.k LIMIT 1)
+----
+anti-join (hash)
+ ├── columns: k:1!null i:2 f:3 s:4 j:5
+ ├── key: (1)
+ ├── fd: (1)-->(2-5)
+ ├── scan a
+ │    ├── columns: k:1!null i:2 f:3 s:4 j:5
+ │    ├── key: (1)
+ │    └── fd: (1)-->(2-5)
+ ├── scan b
+ │    └── columns: y:7
+ └── filters
+      └── y:7 = k:1 [outer=(1,7), constraints=(/1: (/NULL - ]; /7: (/NULL - ]), fd=(1)==(7), (7)==(1)]
+
+# Don't eliminate a non-positive limit.
+norm
+SELECT * FROM a WHERE EXISTS (SELECT * FROM b where b.y = a.k LIMIT 0)
```

Comment:
I'm not sure this test makes sense. On one hand I want EliminateLimitInExists to not affect the expression tree when the limit is non-positive. But the LIMIT expressions ends up being eliminated anyway by another normalization. Thoughts?